yearn/prices/curve.py

```diff
@@ -4,11 +4,11 @@
 
 from yearn.cache import memory
 from yearn.multicall2 import fetch_multicall
+from yearn.constants import CURVE_ADDRESSES_PROVIDER
 
 # curve registry documentation https://curve.readthedocs.io/registry-address-provider.html
-address_provider = Contract('0x0000000022D53366457F9d5E68Ec105046FC4383')
+address_provider = Contract(CURVE_ADDRESSES_PROVIDER)
 curve_registry = Contract(address_provider.get_address(0))
-metapool_factory = Contract(address_provider.get_address(3))
```

Comment:
Don't rely on metapool factory anymore to fetch pool address given an LP address. Instead look at the underlying coins when given an LP address to determine if the LP is actually a metapool

Reply:
I marked the PR as a draft because the method we are using now fetches tokens from the pool rather than underlying tokens. Needs further testing (and refactor) to see if the original approach can still work.

If we choose to use `meta_factory.get_underlying_tokens` we will need to make sure the logic supports bot historical metapool registry implementation and current implementation (current implementation reverts where historic implementation returned ZERO_ADDRESS on failure)